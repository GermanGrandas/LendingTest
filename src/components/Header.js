import { Segment, Menu,Header, Icon,Grid } from "semantic-ui-react";
import React from 'react'
import { Link } from "react-router-dom";

const MenuItems = ({ children, to }) => {
  
    return (
      <Link to={to}
        style={{ color: 'white', marginRight: 15, marginTop: 15 }} >
        <Menu.Item style={{justifyContent: "center", margin:"0.5em 0"}} >
          {children}
        </Menu.Item>
      </Link>
    )
  };

export const HeaderComponent = () => {
    return (
        <header>
            <Segment
                inverted
                // vertical
                // style={{  backgroundColor: "#066732" }}
                textAlign='center'
            >   
                <Menu
                    secondary
                    inverted
                    attached="top"
                    //style={{ display: "flex", height: "15px", backgroundColor: "#066732" }}
                    stackable
                    aling="left"
                >
                    <Grid style={{width : "100%"}}>
                        <Grid.Row columns={3}  centered verticalAlign="middle">
                            <Grid.Column width={6}>
                                <Menu.Menu position='left'>
                                    <img  alt='quantic-logo' src="https://lendingfront.com/wp-content/uploads/2019/12/white_logo.svg"  width={150} height={50} style={{margin:"0.5em 0.4em"}}/>
                                    <MenuItems to='/'><Header as="h4" content="APPLICATION" inverted/></MenuItems>
                                    <MenuItems to='/'>
                                        <Header as="h4" inverted>
                                            <Icon name='search'/>
                                            <Header.Content>SEARCH</Header.Content>
                                        </Header>
                                    </MenuItems>  
                                </Menu.Menu>
                            </Grid.Column>
                            <Grid.Column width={4}>
                                <Header as="h1" content="QA" inverted textAlign='center'  />
                            </Grid.Column>
                            <Grid.Column width={6}>
                                <Menu.Menu position="right">
                            
                                    <MenuItems to='/'><Header as="h4" content="DASHBOARD" inverted/></MenuItems>
                                    <MenuItems to='/'>
                                        <Header as='h4' inverted>
                                            <Icon name='address book' size='small' />
                                            <Header.Content>USER ADMIN</Header.Content>
                                        </Header>
                                    </MenuItems>
                                    <MenuItems to='/'>
                                    <Header as='h4' inverted>
                                        <Icon name='user circle' size='small' />
                                        <Header.Content>USER ADMIN</Header.Content>
                                    </Header>
                                    </MenuItems>
                                </Menu.Menu>
                            </Grid.Column>
                        </Grid.Row>
                    </Grid> 
                    
                </Menu>
            </Segment>
        </header>
    )
}
