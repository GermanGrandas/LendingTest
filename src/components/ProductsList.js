import React, { useRef,useState } from 'react'
import { Grid, Header, Item } from 'semantic-ui-react'

const ItemDetail = ({product, getInvestor}) =>{ 
    const {amount,date,product_id} = product

    const productId = useRef(product_id);
    const [itemRef, setRef] = useState(false);

    const new_date = new Date(date);
    const month = new_date.getMonth() + 1
    const day = new_date.getDate();
    const year = new_date.getFullYear();
    const string_date = `${day}/${month}/${year}`
    const amount_str = `$${amount}`

    const enter = e=> {setRef(true)}
    const leave = e => {setRef(false)}
    
    return (
        <Item className="ui inverted segment item_class" style={{margin : "0 0"}} onMouseOver={enter} onMouseOut={leave} onClick={() => getInvestor(productId.current)}>
            <Item.Content>
            <Item.Header style={{width : "100%"}}>
                <Grid verticalAlign="middle" style={{maxWidth: "100%",margin: "0", padding:"1.4em"}}>
                    <Grid.Row columns={2} style={{padding:"0"}}>
                        <Grid.Column textAlign="left" width={10}>
                            <Grid.Row>
                                <Header as="h6" content="Product ID" inverted={itemRef ? true : false}/>
                            </Grid.Row>
                            <Grid.Row>
                                <Header as="h2" content={product_id} inverted={itemRef ? true : false}/>
                            </Grid.Row>
                        </Grid.Column>
                        <Grid.Column textAlign="right" width={6}>
                            <Grid.Row>
                                <Header as="h5" content="Advance" color={itemRef ? "black" : "teal"} inverted={itemRef ? true : false}/>
                            </Grid.Row>
                            <Grid.Row>
                                <Header as="h6" content={string_date} inverted={itemRef ? true : false}/>
                            </Grid.Row>
                        </Grid.Column>
                    </Grid.Row>
                    
                    <Grid.Row columns={1} style={{padding:"0"}}>
                        <Grid.Column>
                            <Header as="h2" content={amount_str} textAlign="right" inverted={itemRef ? true : false}/>
                        </Grid.Column>
                        
                    </Grid.Row>
                </Grid>
            </Item.Header>
        </Item.Content>
        </Item>
    )
}

export const ProductsList = ({products,getInvestor}) => {
    
    return (
        <Item.Group>
            {
                products.map(product => {
                return (<ItemDetail key={product.product_id} product={product} getInvestor={getInvestor}/>)
            })
            }
        </Item.Group>
    )
}
