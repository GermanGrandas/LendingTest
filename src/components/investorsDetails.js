import React,{useRef,useState} from 'react'
import { Item,Grid,Header, Button, Form } from 'semantic-ui-react'
import { PieChart } from "react-minimal-pie-chart";
import axios from 'axios';



const ItemDetail = ({investor,setEdit}) =>{ 
    const investorId = useRef(investor.id);
    const {investor_name,purchased,sold} = investor
    const sold_str = `$${sold}`
    return (
        <Item className="ui inverted segment simple_border" style={{margin : "0 0"}}>
            <Item.Content>
            <Item.Header style={{width : "100%"}}>
                <Grid verticalAlign="middle" style={{maxWidth: "100%",margin: "0", padding:"1.4em"}}>
                    <Grid.Row columns={4} textAlign="center" style={{padding:"0.1em"}}>
                        <Grid.Column width={5}><Header as="h5" content={investor_name}/></Grid.Column>
                        <Grid.Column width={4}> <Header as="h5" content={sold_str}/></Grid.Column>
                        <Grid.Column width={4}>
                            <PieChart
                                data={[{ value: purchased, color: "#16c4c4" }]}
                                style={{ height: "60px" }}
                                totalValue={100}
                                lineWidth={10}
                                background="#969696"
                                label={({ dataEntry }) => dataEntry.value}
                                labelStyle={{
                                    fontSize: "25px",
                                    fontFamily: "sans-serif",
                                    fill: "#969696",
                                }}
                                labelPosition={0}
                            />
    
                        </Grid.Column>
                        <Grid.Column width={3}>
                            
                            <Button icon="pencil" color="teal" circular size="tiny" onClick={() => setEdit(investorId)}/>    
                            <Button icon="cancel" circular  color="teal" size="tiny" basic/>
                        </Grid.Column>
                    </Grid.Row>
                </Grid>
            </Item.Header>
        </Item.Content>
        </Item>
    )
}

const EditView = ({investors, investor_info, finishEdit}) =>{
    const {investor_name,left_amount,id} = investor_info
    const numeric_amount = parseFloat(left_amount)

    const [formData, setFormData] = useState({id : id, investor: "", sold: left_amount });

    const [error, setError] = useState(false)
    let { investor , sold} = formData
    const changeForm = ({ target }) => {
        setFormData({
          ...formData,
          [target.name]: target.value,
        });
      };
    const handleSelect = (e, { value }) => setFormData({
        ...formData,
        "investor": value,
      });;
    const handleSubmit = e =>{
        e.preventDefault()
        const numerical_pattern = /^[+-]?\d+(\.\d+)?$/;
        const isNumber  = numerical_pattern.test(sold);
        if(isNumber && investor !== ""){
            setError(false)
            axios.put("https://demo7555831.mockable.io/edit_investor",formData).then(
                ({data}) => {
                    
                    finishEdit(formData)
                }
            ).catch(
                err => {
                    setError(true)
                    console.error(err)
                }
            )
            
        }
        else{
            setError(true)
        }
    }
    return(
        <Item className="ui inverted segment simple_border" style={{margin : "0 0"}}>
            <Item.Content>
                <Item.Header style={{width : "100%"}}>
                    <Form onSubmit={handleSubmit} success>
                    <Grid verticalAlign="middle" style={{maxWidth: "100%",margin: "0", padding:"1.4em"}}>
                        
                            <Grid.Row columns={4} textAlign="center" style={{padding:"0.1em"}}>
                                <Grid.Column width={6}>
                                    <Form.Select  fluid
                                        label='Select investor'
                                        options={investors}
                                        onChange={handleSelect}
                                        name="investor"
                                        error={error}
                                        placeholder={investor_name}/>
                                </Grid.Column>
                                <Grid.Column width={4}>
                                    <Form.Input 
                                        fluid label='Amount to sell' 
                                        name="sold" placeholder={numeric_amount} 
                                        onChange={changeForm} value={sold}
                                        error={error}
                                    />
                                </Grid.Column>
                                <Grid.Column width={4}>
                                    <Button icon="save" color="teal" circular size="tiny" type="submit"/>    
                                    <Button icon="cancel" circular  color="teal" size="tiny" basic/>
                                </Grid.Column>
                            </Grid.Row>
                        
                        </Grid>
                    </Form>
                </Item.Header>
            </Item.Content>
        </Item>
    )
}
export const InvestorsDetails = ({investors}) => {
    const [selectedInvestor, setSelectedInvestor] = useState({})
    const [setEditView, setSetEditView] = useState(false)
    const investors_names = investors.map(({investor_name} )=> ({
        key:investor_name,
        value: investor_name,
        text: investor_name,
      }))
    const setEdit = ({current : investor_id }) =>{
        const seleted_inv = investors.filter(({id}) => id === investor_id)[0]
        setSelectedInvestor(seleted_inv)
        setSetEditView(true)
    }

    const finishEdit = investor =>{
        console.log(investor);
        setSetEditView(false)
    }
    return (
        <Item.Group>
            {
                !setEditView ?
                    investors.map(investor => {
                    return (<ItemDetail key={investor.id} investor={investor} setEdit={setEdit}/>)}) :
                <EditView investors={investors_names} investor_info={selectedInvestor} finishEdit={finishEdit}/>
            }
        </Item.Group>
    )
}

export {EditView}