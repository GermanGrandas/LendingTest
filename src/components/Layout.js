import React from 'react'
import { Helmet } from 'react-helmet'
import { Header, Segment } from 'semantic-ui-react'
import { HeaderComponent } from './Header'

export const Layout = ({children}) => {
    const year = new Date().getFullYear() || '2021'
    return (
        <>
          <Helmet>
            <meta charSet="utf-8" />
            <title>LendingTest</title>
            <link rel="canonical" href="https://germangrandas.github.io/LendingTest/" />
          </Helmet>
         <HeaderComponent/>
         <main>{children}</main>
         <footer >
             
         <Segment
          vertical
          
          //color="black"
          size='mini'
          textAlign='center'
          style={{
            padding: "4em 0em",
            boxShadow: "-8px 15px 30px rgba(133,141,148,.66)",
            // backgroundColor: "#000",
            //border: "1px solid black",
          }}
        >
            <Header as="h3" content={`Â©LendingFront${year}`}/>
        </Segment>
         </footer>
        </>
    )
}
