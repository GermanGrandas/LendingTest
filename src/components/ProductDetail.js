import React, {useState,useEffect} from 'react'
import { Grid, Header, Progress } from 'semantic-ui-react'
import axios from 'axios';
import { InvestorsDetails } from './investorsDetails';

const ProductDetail = ({product}) => {
    const [information, setInformation] = useState([])
    useEffect(() => {
        axios.post(
            "https://demo7555831.mockable.io/get_investors_by_id",{product_id : product}
        ).then(
            ({data}) => {
                let {information} = data
                setInformation(information)
            }
        ).catch(console.error)
    }, [product])
    return (
        <Grid columns={1}>
            <Grid.Column>
                <Grid.Row className="investors_space">
                    <Grid.Row style={{margin: "0.3em 1em",}}>
                        <Header as="h4" content={`Product ID ${product}`} 
                            inverted className="border_header"/>
                    </Grid.Row>
                    <Grid.Row style={{marginTop: "1.3em"}} >
                        <Grid verticalAlign="middle">
                            <Grid.Row columns={4} textAlign="center">
                                <Grid.Column width={5}><Header as="h5" content="Investor Name" inverted/></Grid.Column>
                                <Grid.Column width={4}> <Header as="h5" content="Sold" inverted/></Grid.Column>
                                <Grid.Column width={4}><Header as="h5" content="% Purchased" inverted/></Grid.Column>
                                <Grid.Column width={3}></Grid.Column>
                            </Grid.Row>
                        </Grid>
                    </Grid.Row>
                </Grid.Row>
                <Grid.Row style={{overflowY : "scroll",maxHeight : "43.5vh", minHeight : "43.5vh"}}>
                    <InvestorsDetails investors={information}/>
                </Grid.Row>
                <Grid.Row className="card_button teal_space" verticalAlign="middle" textAlign="center">
                    <Header as="h3" style={{margin : "0.4em  1.4em"}}
                        content="Remaining amount $850.000 of $8.300.800" inverted/>
                    <Progress percent={25} style={{margin : "0.4em  1.4em",}} color="blue"
                        size='tiny'/>
                    <Grid.Row style={{margin : "0.4em  1.4em",}}>
                        <Header as="h6" content="0%" floated='left' inverted/>
                        <Header as="h6" content="100%" floated="right" inverted/>
                    </Grid.Row>
                    
                </Grid.Row>
            </Grid.Column>
            
        </Grid>
    )
}
export default ProductDetail