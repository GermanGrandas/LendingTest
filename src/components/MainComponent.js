import React, {useState,useEffect} from 'react'
import { Button, Grid,Header,Icon,Segment } from 'semantic-ui-react'
import axios from 'axios';

import ProductDetail from './ProductDetail';
import ProductsList from './ProductsList';

// import { data_products } from "../staticInfo";

export const MainComponent = () => {
    const [information, setInformation] = useState([])
    const [currentProduct, setCurrentProduct] = useState("")

    useEffect(() => {
        axios.post(
            "https://demo7555831.mockable.io/get_products_by_id",{user_id : ""}
        ).then(
            ({data}) => {
                let {information} = data
                setInformation(information)
            }
        ).catch(console.error)
    }, [])
    
    const getInvestor = productId =>{
        setCurrentProduct(productId)
    }
    return (
        <Segment attached="top" style={{ padding : "0 0", height : "100%"}}>
            <Grid columns={2} doubling stretched style={{margin : "0 0",marginLeft : "0 !important", height : "100%"}}>
                <Grid.Column width={5} style={{ height : "100%",padding : "0 0"}}>
                    <div className="ui left aligned card_header">
                        <Header as="h4" content="Select a product to syndicate" style={{margin : "0.5em 2em"}}/>
                    </div>
                    <div style={{overflowY : "scroll",minHeight : "48vh"}}>
                        <ProductsList products={information} getInvestor={getInvestor} selected={currentProduct}/>
                    </div>
                    
                    <Segment textAlign="center" className="card_button">
                        <Grid columns={2}>
                            <Grid.Column>
                                <Button color='teal' size="medium" style={{borderRadius : "15px", width : "160px"}}>
                                    <Icon name="close" /> Close
                                </Button>
                            </Grid.Column>
                            <Grid.Column>
                                <Button size='medium' 
                                    style={{ borderRadius : "15px", width : "160px", 
                                             background :"white" , color :"teal",
                                             border : "1px solid rgba(228, 228, 228,0.6)"}}>
                                    <Icon name="save" /> Sell
                                </Button>
                            </Grid.Column>
                        </Grid>
                        
                        
                    </Segment>
                    
                </Grid.Column>
                <Grid.Column width={11} style={{padding : "0"}}>
                    <ProductDetail product={currentProduct}/>
                </Grid.Column>
            </Grid>
        </Segment>
        
    )
}
