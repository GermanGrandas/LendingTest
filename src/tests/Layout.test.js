import React from 'react';
import {shallow} from 'enzyme'
import {Layout} from "../components/Layout";

describe('<Layout /> component test', () => {
    test('Layout Snapshot', () => {
        
        const wrapper = shallow(<Layout />)

        expect(wrapper).toMatchSnapshot();
    })
    
})