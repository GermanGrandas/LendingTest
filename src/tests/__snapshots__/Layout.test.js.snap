// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> component test Layout Snapshot 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <title>
      LendingTest
    </title>
    <link
      href="https://germangrandas.github.io/LendingTest/"
      rel="canonical"
    />
  </HelmetWrapper>
  <HeaderComponent />
  <main />
  <footer>
    <Segment
      size="mini"
      style={
        Object {
          "boxShadow": "-8px 15px 30px rgba(133,141,148,.66)",
          "padding": "4em 0em",
        }
      }
      textAlign="center"
      vertical={true}
    >
      <Header
        as="h3"
        content="Â©LendingFront2021"
      />
    </Segment>
  </footer>
</Fragment>
`;
